src/txmempool.h

```diff
@@ -62,6 +68,27 @@ class CTxMemPoolEntry
     size_t DynamicMemoryUsage() const { return nUsageSize; }
 };
 
+// extracts a TxMemPoolEntry's transaction hash
+struct mempoolentry_txid
+{
+    typedef uint256 result_type;
+    result_type operator() (const CTxMemPoolEntry &entry) const
+    {
+        return entry.GetTx().GetHash();
+    }
+};
+
+class CompareTxMemPoolEntryByFee
```

Comment:
NACK

Any comparator is ultimately going to be about profit-per-byte. (modulo radical design changes to how the Bitcoin protocol works)